### PR TITLE
[CodeCamp] Add doc of `module_combination`

### PR DESCRIPTION
**Pull request by kitecats**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+### 替换MMYOLO Head中的loss
+
+OpenMMLab 2.0 体系中 MMYOLO、MMDetection、MMClassification 中的loss注册表都继承自 MMEngine 中的根注册表。 因此用户可以在 MMYOLO 中使用来自 MMDetection、MMClassification 中实现的loss而无需重新实现。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:3]:
```suggestion
OpenMMLab 2.0 体系中 MMYOLO、MMDetection、MMClassification 中的 loss 注册表都继承自 MMEngine 中的根注册表。 因此用户可以在 MMYOLO 中使用来自 MMDetection、MMClassification 中实现的loss而无需重新实现。
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+### 替换MMYOLO Head中的loss
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:1]:
```suggestion
### 替换 MMYOLO Head 中的 loss
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+OpenMMLab 2.0 体系中 MMYOLO、MMDetection、MMClassification 中的loss注册表都继承自 MMEngine 中的根注册表。 因此用户可以在 MMYOLO 中使用来自 MMDetection、MMClassification 中实现的loss而无需重新实现。
+
+#### 使用实现好的损失函数替换 Head 中的 loss_cls
+
+1. 假设我们想使用 LabelSmoothLoss 作为 loss_cls 的损失函数。因为 LabelSmoothLoss 已经在 MMClassification 中实现了，所以可以直接在配置文件中进行替换。配置文件如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:7]:
```suggestion
1. 假设我们想使用 `LabelSmoothLoss` 作为 `loss_cls` 的损失函数。因为 `LabelSmoothLoss` 已经在 MMClassification 中实现了，所以可以直接在配置文件中进行替换。配置文件如下：
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        _delete_=True,
+        _scope_='mmcls', #  临时替换scope为mmcls
+        type='LabelSmoothLoss',
+        label_smooth_val=0.1,
+        mode='multi_label',
+        reduction='mean',
+        loss_weight=0.5)))
+```
+
+2. 假设我们想使用 VarifocalLoss 作为 loss_cls 的损失函数。因为 VarifocalLoss 在 MMDetection 已经实现好了，所以可以直接替换。配置文件如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:24]:
```suggestion
2. 假设我们想使用 `VarifocalLoss` 作为 `loss_cls` 的损失函数。因为 `VarifocalLoss` 在 MMDetection 已经实现好了，所以可以直接替换。配置文件如下：
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+model = dict(
+    bbox_head=dict(
+        loss_cls=dict(
+            _delete_=True,
+            _scope_='mmdet',
+            type='VarifocalLoss',
+            loss_weight=1.0)))
+```
+
+3. 假设我们想使用 FocalLoss 作为 loss_cls 的损失函数。配置文件如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:36]:
```suggestion
3. 假设我们想使用 `FocalLoss` 作为 `loss_cls` 的损失函数。配置文件如下：
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+model = dict(
+    bbox_head=dict(
+        loss_cls= dict(
+            _delete_=True,
+            type='FocalLoss',
+            loss_weight=1.0)))
+```
+
+4. 假设我们想使用 QualityFocalLoss 作为 loss_cls 的损失函数。配置文件如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:48]:
```suggestion
4. 假设我们想使用 `QualityFocalLoss` 作为 `loss_cls` 的损失函数。配置文件如下：
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    bbox_head=dict(
+      loss_cls= dict(
+        _delete_=True,
+        type='QualityFocalLoss',
+        loss_weight=1.0)))
+```
+
+#### 使用实现好的损失函数替换 Head 中的 loss_obj
+
+loss_obj 的替换与 loss_cls 的替换类似，我们可以使用已经实现好的损失函数对 loss_obj 的损失函数进行替换
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:62]:
```suggestion
`loss_obj` 的替换与 `loss_cls` 的替换类似，我们可以使用已经实现好的损失函数对 `loss_obj` 的损失函数进行替换
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+1. 假设我们想使用 VarifocalLoss 作为 loss_obj 的损失函数
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:64]:
```suggestion
1. 假设我们想使用 `VarifocalLoss` 作为 `loss_obj` 的损失函数
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+model = dict(
+    bbox_head=dict(
+        loss_obj=dict(
+            _delete_=True,
+            _scope_='mmdet',
+            type='VarifocalLoss',
+            loss_weight=1.0)))
+```
+
+2. 假设我们想使用 FocalLoss 作为 loss_obj 的损失函数。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:76]:
```suggestion
2. 假设我们想使用 `FocalLoss` 作为 `loss_obj` 的损失函数。
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+model = dict(
+    bbox_head=dict(
+        loss_cls= dict(
+            _delete_=True,
+            type='FocalLoss',
+            loss_weight=1.0)))
+```
+
+3. 假设我们想使用 QualityFocalLoss 作为 loss_obj 的损失函数。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:88]:
```suggestion
3. 假设我们想使用 `QualityFocalLoss` 作为 `loss_obj` 的损失函数。
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+      loss_cls= dict(
+        _delete_=True,
+        type='QualityFocalLoss',
+        loss_weight=1.0)))
+```
+
+#### 注意
+
+1. 损失函数的替换只是在理论上的替换，不能保证实际效果会有所提升。
+2. 本次损失函数的替换都是以 yolov5 算法作为例子的，但是 MMYOLO 下的多个算法，如 yolov6，yolox 等算法都可以按照上述的例子进行替换。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:103]:
```suggestion
2. 本次损失函数的替换都是以 YOLOv5 算法作为例子的，但是 MMYOLO 下的多个算法，如 YOLOv6，YOLOX 等算法都可以按照上述的例子进行替换。
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
+### 替换MMYOLO Head中的 loss
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:1]:
```suggestion
# 算法组合替换教程
## Loss 组合替换教程
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+
+OpenMMLab 2.0 体系中 MMYOLO、MMDetection、MMClassification 中的 loss 注册表都继承自 MMEngine 中的根注册表。 因此用户可以在 MMYOLO 中使用来自 MMDetection、MMClassification 中实现的loss而无需重新实现。
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:3]:
```suggestion
OpenMMLab 2.0 体系中 MMYOLO、MMDetection、MMClassification 中的 loss 注册表都继承自 MMEngine 中的根注册表。 因此用户可以在 MMYOLO 中使用来自 MMDetection、MMClassification 中实现的 loss 而无需重新实现。
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+
+#### 使用实现好的损失函数替换 Head 中的 loss_cls
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:5]:
```suggestion
### 替换 YOLOv5 Head 中的 loss_cls 函数
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+1. 假设我们想使用 `LabelSmoothLoss` 作为 `loss_cls` 的损失函数。因为 `LabelSmoothLoss` 已经在 MMClassification 中实现了，所以可以直接在配置文件中进行替换。配置文件如下：
+
+```python
+# 请先使用命令： mim install "mmcls>=1.0.0rc2"，安装 mmcls
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+model = dict(
+    bbox_head=dict(
+      loss_cls=dict(
+        _delete_=True,
+        _scope_='mmcls', #  临时替换scope为mmcls
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:16]:
```suggestion
        _scope_='mmcls', #  临时替换 scope 为 mmcls
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+model = dict(
+    bbox_head=dict(
+      loss_cls= dict(
+        _delete_=True,
+        type='QualityFocalLoss',
+        loss_weight=1.0)))
+```
+
+#### 使用实现好的损失函数替换 Head 中的 loss_obj
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:60]:
```suggestion
### 替换 YOLOV5 Head 中的 loss_obj 函数
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+    bbox_head=dict(
+      loss_cls= dict(
+        _delete_=True,
+        type='QualityFocalLoss',
+        loss_weight=1.0)))
+```
+
+#### 注意
+
+1. 损失函数的替换只是在理论上的替换，不能保证实际效果会有所提升。
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:102]:
```suggestion
1. 在本教程中损失函数的替换是运行不报错的，但无法保证性能一定会上升。
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+2. 假设我们想使用 `FocalLoss` 作为 `loss_obj` 的损失函数。
+
+```python
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+model = dict(
+    bbox_head=dict(
+        loss_cls= dict(
+            _delete_=True,
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:83]: 漏了 scope

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+3. 假设我们想使用 `QualityFocalLoss` 作为 `loss_obj` 的损失函数。
+
+```python
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+model = dict(
+    bbox_head=dict(
+      loss_cls= dict(
+        _delete_=True,
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:95]: 漏了 scope

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**